src/main/java/com/tozny/sdk/RealmApi.java

```diff
@@ -247,10 +247,11 @@ private boolean dispatchUserExists(UserExistsRequest req) throws ToznyApiExcepti
     }
 
     private boolean dispatchUserPush(UserPushRequest request) throws ToznyApiException {
-        ToznyApiResponse<Boolean> response = protocol.<ToznyApiResponse<Boolean>>dispatch(
-                request, new TypeReference<ToznyApiResponse<Boolean>>() {});
-        String ret = response.getReturn();
-        return ret != null && ret.equals("true");
+        ToznyApiResponse<Boolean[]> response = protocol.<ToznyApiResponse<Boolean[]>>dispatch(
+                request, new TypeReference<ToznyApiResponse<Boolean[]>>() {});
+
+        Boolean[] results = response.getResult();
+        return results != null && results.length == 1 && results[0];
```

Comment:
If there's an exception, it's caught and cast as a `TonzyApiException`. All other returns from this method are intended to just be `false` to indicate the push did not go out. The API itself literally only returns `true` or `false` and doesn't indicate _why_ the push failed, just that it failed.